backend/test/unit/utils/token.test.ts

```diff
@@ -0,0 +1,304 @@
+import { ChatCompletionMessageParam } from 'openai/resources/chat/completions';
+
+import { filterChatHistoryByMaxTokens } from '@src/utils/token';
+
+describe('token unit tests', () => {
+	test('GIVEN chat history exceeds max token number WHEN applying filter THEN it should return the filtered chat history', () => {
+		const maxTokens = 260; // 120 tokens for function call
```

Comment:
this comment confuses me

Reply:
the model is set up with function call definitions (or that is chatGptTools) which uses up 120 tokens before any thing is added to the chat history. 
So if maxTokens = 260, 120 are already used for the function calls, then the rest can be used in chat history. 

I will update the comments to make this clearer.

Reply:
ah, I see!